src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+2. cd RubricEval/ && pip install -e .
+3. python scripts/convert_dataset_to_rubriceval_input_format.py  # This will generate an example instructions.json
+4. rubric_eval get_rubrics --input_path=instructions.json --output_path=instructions_with_rubrics.json
+5. rubric_eval get_completions --model_configs="['gpt-4o-2024-05-13']" --input_path=instructions_with_rubrics.json --output_path_prefix=output_
+6. rubric_eval --input_path_prefix=output_ --output_path_prefix=eval_output_
+"""
+
+
+import sys
+import os
+import ast
+from pathlib import Path
+import pandas as pd
+from typing import List
+from alpaca_eval import utils as ae_utils
+
+import fire
+
+from .helper import get_detailed_rubrics, get_model_completions, get_evaluations
+from .annotators import RubricBrainstormer
+
+CUR_DIR = Path(__file__).parent
+
+__all__ = ["get_rubrics", "get_outputs", "eval_outputs"]
+
+
+def check_df_fields(df, required_fields, optional_fields=set()):
+    allowed_fields = required_fields | optional_fields
+    
+    actual_fields = set(df.columns)
+    
+    # Check if all required fields are present
+    if not required_fields.issubset(actual_fields):
+        missing = required_fields - actual_fields
+        raise RuntimeError(f"Missing required fields: {missing}")
+    
+    # Check if there are any fields not in the allowed set
+    extra_fields = actual_fields - allowed_fields
+    if extra_fields:
+        raise RuntimeError(f"Extra fields found: {extra_fields}")
+    
+
+def cleanup_df_rubrics(df_rubrics):
+    # Drop rows where the "scoring_scales" column is not a dict (because sometimes rubric generator will refuse to generate rubric)
+    mask = df_rubrics["scoring_scales"].apply(lambda x: isinstance(x, dict))
+    return df_rubrics[mask]
+
+
+def list_files_with_prefix(path_prefix, suffix):
+    matching_files = []
+    
+    # Expand path_prefix to full absolute path
+    full_path_prefix = os.path.abspath(path_prefix)
+    
+    # Get the directory part of the full_path_prefix
+    prefix_dir = os.path.abspath(os.path.dirname(full_path_prefix))
+    
+    # Walk through the directory
+    for root, _, files in os.walk(prefix_dir):
+        for file in files:
+            full_path = os.path.join(root, file)
+            if (
+                full_path == os.path.join(prefix_dir, file)
+                and full_path.startswith(full_path_prefix)
+                and full_path.endswith(suffix)
+            ):
+                matching_files.append(full_path)
+    
+    return matching_files
+
+
+def string_to_list(s):
+    try:
+        return ast.literal_eval(s)
+    except (ValueError, SyntaxError):
+        # Handle the case where the string is not a valid literal
+        print(f"Error: '{s}' is not a valid list representation")
+        return None
+
+
+
+def get_rubrics(
+    input_path: str = "instructions.json",
+    output_path: str = "instructions_with_rubrics.json",
+):
+    """
+    input_path: str, path to input file (JSON)
```

Comment:
let's use numpy or google style docstrings. I have a slight preference for numpy to be consistent with alpaca_eval but either is fine for me as long as we always use the same in the codebase: your choice!

It's important because IDEs will nicely retrieve such docstrings

Reply:
see: https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy